src/plans/constrained_plan.jl

```diff
@@ -35,824 +12,837 @@ Describes the constrained objective
 
-evaluate all gradients of the inequality constraints ``\operatorname{grad} g(x)`` or ``\bigl(\operatorname{grad} g_1(x), \operatorname{grad} g_2(x),\ldots,\operatorname{grad} g_m(x)\bigr)``
-of the [`ConstrainedManifoldObjective`](@ref) `P` at `p` in place of `X`, which is a vector of ``m`` tangent vectors.
-"""
-function get_grad_inequality_constraints!(
-    M::AbstractManifold,
-    X,
-    co::ConstrainedManifoldObjective{AllocatingEvaluation,FunctionConstraint},
+function get_hess_inequality_constraint!(
+    amp::AbstractManoptProblem,
+    Y,
     p,
-)
-    copyto!.(Ref(M), X, Ref(p), co.grad_g!!(M, p))
-    return X
-end
-function get_grad_inequality_constraints!(
-    M::AbstractManifold,
     X,
-    co::ConstrainedManifoldObjective{AllocatingEvaluation,VectorConstraint},
-    p,
+    j=:,
+    range::AbstractPowerRepresentation=NestedPowerRepresentation(),
 )
-    for (Xi, grad_gi) in zip(X, co.grad_g!!)
-        copyto!(M, Xi, grad_gi(M, p))
-    end
-    return X
+    return get_hess_inequality_constraint!(
+        get_manifold(amp), Y, get_objective(amp), p, X, j, range
+    )
 end
-function get_grad_inequality_constraints!(
-    M::AbstractManifold,
-    X,
-    co::ConstrainedManifoldObjective{InplaceEvaluation,FunctionConstraint},
-    p,
+function get_hess_inequality_constraint!(cmp::ConstrainedManoptProblem, Y, p, X, j=:)
+    return get_hess_inequality_constraint!(
+        get_manifold(cmp), Y, get_objective(cmp), p, X, j, cmp.hess_ineqality_range
+    )
+end
+function get_hess_inequality_constraint!(
+    M::AbstractManifold, Y, admo::AbstractDecoratedManifoldObjective, args...
 )
-    co.grad_g!!(M, X, p)
-    return X
+    return get_hess_inequality_constraint!(M, Y, get_objective(admo, false), args...)
 end
-function get_grad_inequality_constraints!(
+function get_hess_inequality_constraint!(
     M::AbstractManifold,
-    X,
-    co::ConstrainedManifoldObjective{InplaceEvaluation,VectorConstraint},
+    Y,
+    co::ConstrainedManifoldObjective,
     p,
+    X,
+    j=:,
+    range::AbstractPowerRepresentation=NestedPowerRepresentation(),
 )
-    for (Xi, grad_gi!) in zip(X, co.grad_g!!)
-        grad_gi!(M, Xi, p)
-    end
-    return X
+    isnothing(co.equality_constraints) && (return X)
+    return get_hessian!(M, Y, co.inequality_constraints, p, X, j, range)
 end
-function get_grad_inequality_constraints!(
-    M::AbstractManifold, X, admo::AbstractDecoratedManifoldObjective, p
-)
-    return get_grad_inequality_constraints!(M, X, get_objective(admo, false), p)
+
+@doc raw"""
+    inequality_constraints_length(co::ConstrainedManifoldObjective)
+
+Return the number of equality constraints of an [`ConstrainedManifoldObjective`](@ref).
+This acts transparently through [`AbstractDecoratedManifoldObjective`](@ref)s
+"""
+function inequality_constraints_length(co::ConstrainedManifoldObjective)
+    return length(co.inequality_constraints)
```

Comment:
I think that won't work for vectorized constraints?

Reply:
I do not follow? `co.inequality_constraints` is _always_ a `VectorGradientFunction` or a `VectorHessianFunction` – for both types`length` is implemented.

Reply:
I see, then maybe let's restrict `EMO` and `IMO` to `AbstractVectorGradientFunction` for clarity?

Reply:
Yes that sounds reasonable.

Reply:
It turns out either `inequality_constraints` or `equality_constraints` can be `nothing`, in which case `length` won't work.

Reply:
then we could return zero here.

Reply:
I have a meeting in a moment and other work so I most likely won't be working on this today.

Reply:
No problem I can add that line and hope to find time to check the error in the docs.

Reply:
Fixed that.